src/olympia/ratings/views.py

```diff
@@ -412,6 +433,7 @@ def get_paginated_response(self, data):
                     'the user id of the authenticated user')
             extra_data['can_reply'] = (
                 self.check_can_reply_permission_for_ratings_list())
+        self.should_include_flags()
```

Comment:
Why do we need that here ?

Reply:
`should_include_flags` has the validation in it so if we don't call it here (or somewhere else in the view explicitly) then when there aren't any results to serialize an invalid request (e.g. they aren't authenticated, the `show_flags_for` parameter doesn't match the user) isn't raised in the API response.

This test https://github.com/mozilla/addons-server/pull/9988/files/f4bf3d514e65fb61a6dafafc552539de0bcfadc3#diff-dbe5b121611cfb846d3adfc78246f12aR1309 and the other two tests on invalid list requests fail.

Reply:
Ok, that'll need a comment :)

Reply:
https://github.com/mozilla/addons-server/pull/9988/commits/a844a565ea9d2169cffae6d6a5404e2e9e3717ce